pkg/devcontainer/image_manager.go

```diff
@@ -0,0 +1,147 @@
+package devcontainer
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/artmoskvin/hide/pkg/util"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/rs/zerolog/log"
+)
+
+type ImageManager interface {
+	PullOrBuildImage(ctx context.Context, workingDir string, config Config) (string, error)
+	PullImage(ctx context.Context, name string) error
+	BuildImage(ctx context.Context, buildContextPath, dockerFilePath string, buildProps *BuildProps, containerName string) (string, error)
+}
+
+type DockerImageManager struct {
+	dockerClient DockerClient
+	config       DockerRunnerConfig
+}
+
+func NewImageManager(dockerClient DockerClient, config DockerRunnerConfig) ImageManager {
+	return &DockerImageManager{dockerClient: dockerClient, config: config}
+}
+
+func (im *DockerImageManager) PullOrBuildImage(ctx context.Context, workingDir string, config Config) (string, error) {
```

Comment:
I thought you did not like `Or` methods. What is the rational for it?

Reply:
TODO:
- split into separate methods
- create aux methods in devcon.Config to define config type

Reply:
Done